### ISSUE TITLE
KubeEdge supports Istio?

### ISSUE THREAD

**Issue by fujitatomoya** [kubeedge/kubeedge#3846]
This is general question if KubeEdge supports Istio.
I can see `Unified Service Mesh support (Integrate with Istio/OSM etc.)` in [roadmap](https://kubeedge.io/en/docs/roadmap) 2021, but could not find the official support announcement.

there are a couple of related questions as below.

- https://github.com/kubeedge/kubeedge/issues/2314
- https://github.com/kubeedge/kubeedge/issues/2677

After all, can we deploy Istio with KubeEdge?

**Comment by fisherxu** [comment]
@fujitatomoya Yes, now we can deploy Istio with KubeEdge, but you need to ensure the edge nodes can connect to each other. And could you please provide more details that how do you want to use istio in edge? :)

**Comment by fujitatomoya** [comment]
@fisherxu awesome! and thanks for the quick response!

> you need to ensure the edge nodes can connect to each other.

Is this supposed to mean that all edge nodes running edgecore, they all must be accessible via network? i think it is unlikely that edge devices are in the same network... in that case, we need to configure edge nodes to connect each other with using like VPN or something.
Or does that mean Pods/Containers on edge nodes need to be accessible via CNI? according to [roadmap](https://kubeedge.io/en/docs/roadmap/), `Integration and verification of third-party CNI` CNI could be bound on edge Pods? if so, we could use WeaveNet to connect all edge Pods (including cloud Pods) and establish Istio over KubeEdge/Kubernetes?

> And could you please provide more details that how do you want to use istio in edge?

we are looking at edge and cloud common infrastructure as concept. probably you could see the idea [KubeCon EU Edge Day 2021 Presentation](https://static.sched.com/hosted_files/kubenetesedgedayeu21/36/KubernetesRoboticsEdgeClusterSystem_TomoyaFujita_4May_v1.pdf).
